torch/_inductor/kernel/flex_attention.py

```diff
@@ -615,7 +615,13 @@ def _get_default_config_fwd(query) -> Tuple[int, int, int, int]:
     head_dim = query.get_size()[-1]
     default_config = None
 
-    if head_dim <= 256 and torch.cuda.get_device_capability() >= (9, 0):  # H100
+    if head_dim <= 256 and torch.version.hip:
```

Comment:
I think it might be cleaner to instead create a _get_config_rocm and we branch at call site

TBH this flow is already hard to grok

Reply:
Do you have any specific recommendations @drisspg

Reply:
I think we should copy and paste the entire function and create two versions, 1 for rocm and for cuda 1. It feel to me although you need branching on head_dms that the branching should likely be different given the reduced smem constraints and increased warp size on AMD gpus. 

I think lets just make two funcs and copy and paste as much as you need, until the dust settles some more